File: app/src/main/java/com/example/blackice/LoginActivity.java
package com.example.blackice;

import androidx.appcompat.app.AppCompatActivity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import android.widget.Toast;


public class LoginActivity extends AppCompatActivity {
    private Button Login;
    private Button Signin;
    private EditText LoginId;
    private EditText LoginPw;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        Signin = (Button) findViewById(R.id.SigninBtn);
        Login = (Button) findViewById(R.id.LoginBtn);
        LoginId = (EditText) findViewById(R.id.userId);
        LoginPw = (EditText) findViewById(R.id.userPw);

        Signin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // SignipActivity 연결 - 예현 만드는 중(7/9)
                Intent intent = new Intent(LoginActivity.this, SignInActivity.class);
                startActivity(intent);
            }
        });

        Login.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String userid = LoginId.getText().toString().trim();
                String userpw = LoginPw.getText().toString().trim();
                /* 파이어베이스 연결 부분 ,
                 firebaseAuth.signInWithEmailAndPassword(userid,userpw)
                        .addOnCompleteListener(MainActivity.this, new OnCompleteListener<AuthResult>() {
                            @Override
                            public void onComplete(@NonNull Task<AuthResult> task) {
                                if(task.isSuccessful()){
                                    Intent intent = new Intent(MainActivity.this, BottomNaviActivity.class);
                                    startActivity(intent);

                                }else{
                                    Toast.makeText(MainActivity.this,"로그인 오류",Toast.LENGTH_SHORT).show();
                                }
                 */

            }
        });



    }








}

File: app/src/main/java/com/example/blackice/LoginActivity.kt
package com.example.blackice

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.ValueEventListener
import com.google.firebase.database.ktx.database
import com.google.firebase.ktx.Firebase
import kotlinx.android.synthetic.main.activity_login.*

var id=""
var pwd=""
class LoginActivity : AppCompatActivity() {
    var firebaseRef = Firebase.database.getReference("App")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        SigninBtn.setOnClickListener(View.OnClickListener { // SignipActivity 연결 - 예현 만드는 중(7/9)
            val intent = Intent(this@LoginActivity, SignInActivity::class.java)
            startActivity(intent)
        })
        LoginBtn.setOnClickListener(View.OnClickListener {
            id =userId.text.toString()
            pwd = userPw.text.toString()

            firebaseRef.child("Users").addListenerForSingleValueEvent(object : ValueEventListener {
                override fun onDataChange(snapshot: DataSnapshot) {
                    val children = snapshot.children.iterator()
                    var check = false
                    var key:String?
                    while (children.hasNext())
                    {
                        key = children.next().key
                        if(key==id)
                        {
                            firebaseRef.child("Users").child(key).child("Userpwd").addListenerForSingleValueEvent(object : ValueEventListener {
                                override fun onDataChange(snapshot: DataSnapshot) {
                                    val pw=snapshot.value.toString()

                                    if(pwd==pw )
                                    {
                                        val intent = Intent(this@LoginActivity, MainActivity::class.java)
                                        intent.putExtra("userid", id)
                                        startActivity(intent)
                                        return

                                    }
                                    else
                                    {
                                        toast()

                                    }

                                }

                                override fun onCancelled(error: DatabaseError) {
                                    println("Failed to read value.")
                                }
                            })

                            check=true
                        }
                        else if(key!=id && !children.hasNext() && !check)
                        {
                            toast()

                        }
                    }

                }



                override fun onCancelled(error: DatabaseError) {
                    println("Failed to read value.")
                }
            })

        })
    }
    private fun toast() {
        Toast.makeText(this, "ID 혹은 비밀번호가 일치하지 않습니다.", Toast.LENGTH_SHORT).show()
    }
}